Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
```

Comment:
Nice :) But you do need to have C++ background to perceive this as natural :)
I remember in C++ you could have multiple values inside of the round brackets, and the last one would get returned: (a, b, c)  :)